internal/cli/opsmanager/clusters/delete.go

```diff
@@ -43,39 +48,162 @@ func (opts *DeleteOpts) Run() error {
 	if err != nil {
 		return err
 	}
 
-	if !search.ClusterExists(current, opts.Entry) {
+	for _, host := range hosts.Results {
+		if ports, ok := hostname[host.Hostname]; ok {
+			for _, port := range ports {
+				if port == host.Port {
+					opts.hostIds = append(opts.hostIds, host.ID)
+				}
+			}
+		}
+	}
+
+	if len(opts.hostIds) == 0 {
 		return fmt.Errorf("cluster '%s' doesn't exist", opts.Entry)
 	}
 
+	return nil
+}
+
+func (opts *DeleteOpts) stopMonitoring() error {
+	for _, id := range opts.hostIds {
+		err := opts.store.StopMonitoring(opts.ConfigProjectID(), id)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (opts *DeleteOpts) removeClusterFromAutomation() error {
+	current, err := opts.store.GetAutomationConfig(opts.ConfigProjectID())
+	if err != nil {
+		return err
+	}
+
 	atmcfg.RemoveByClusterName(current, opts.Entry)
+	if err := opts.store.UpdateAutomationConfig(opts.ConfigProjectID(), current); err != nil {
+		return err
+	}
 
+	// Wait for changes being deployed on automation
+	if err := opts.watch(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (opts *DeleteOpts) shutdownCluster() error {
+	current, err := opts.store.GetAutomationConfig(opts.ConfigProjectID())
+	if err != nil {
+		return err
+	}
+	if !search.ClusterExists(current, opts.Entry) {
+		return fmt.Errorf("cluster '%s' doesn't exist", opts.Entry)
+	}
+
+	// Shutdown Cluster
+	atmcfg.Shutdown(current, opts.Entry)
 	if err := opts.store.UpdateAutomationConfig(opts.ConfigProjectID(), current); err != nil {
 		return err
 	}
 
-	fmt.Print(cli.DeploymentStatus(config.OpsManagerURL(), opts.ConfigProjectID()))
+	// Wait for changes being deployed on automation
+	if err := opts.watch(); err != nil {
+		return err
+	}
 
 	return nil
 }
 
-// mongocli cloud-manager cluster(s) delete <name> --projectId projectId [--force]
+func (opts *DeleteOpts) watch() error {
+	for {
+		result, err := opts.watcher()
+		if err != nil {
+			return err
+		}
+		if !result {
+			time.Sleep(defaultWait)
+		} else {
+			return nil
+		}
+	}
+}
+
+func (opts *DeleteOpts) watcher() (bool, error) {
+	result, err := opts.store.GetAutomationStatus(opts.ConfigProjectID())
+	if err != nil {
+		return false, err
+	}
+
+	for _, p := range result.Processes {
+		if p.LastGoalVersionAchieved != result.GoalVersion {
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+// mongocli cloud-manager cluster(s) unmanage <name> --projectId projectId [--force]
 func DeleteBuilder() *cobra.Command {
 	opts := &DeleteOpts{
 		DeleteOpts: cli.NewDeleteOpts("", "Cluster not deleted\""),
 	}
 	cmd := &cobra.Command{
-		Use:     "delete <name>",
-		Aliases: []string{"rm"},
-		Short:   "This is an internal and undocumented command.",
-		Long:    "This commands only removes entries from the automation config but does not actually remove a cluster.",
+		Use:     "unmanage <name>",
+		Aliases: []string{"rm", "delete"},
+		Short:   DeleteCluster,
```

Comment:
sorry if this was not clear from the ticket, we want to keep the old delete. the old delete should be renamed to `unmanage`, users may still want to just unmanaged a deployment
Then we want a new `delete` command, which does what you just implemented